### PR TITLE
[RayService] make RayClusterSpec required

### PR DESCRIPTION
**Pull request by fscnick**
## Why are these changes needed?

To make `RayClusterSpec` required in RayService.

This is the support PR according to https://github.com/ray-project/kuberay/pull/3153#discussion_r1983802799

Here is the result after applying a RayService with an empty `RayClusterSpec`.
fail

## Checks

- [x] I've made sure the tests are passing.
- Testing Strategy
  - [x] Unit tests
  - [ ] Manual tests
  - [ ] This PR is not tested :(

### CODE REVIEW

File: ray-operator/controllers/ray/utils/validation_test.go [hunk 1]
```diff
@@ -782,7 +782,19 @@ func TestValidateRayServiceSpec(t *testing.T) {
...
 
 	err = ValidateRayServiceSpec(&rayv1.RayService{
-		Spec: rayv1.RayServiceSpec{},
+		Spec: rayv1.RayServiceSpec{
+			RayClusterSpec: rayv1.RayClusterSpec{
+				HeadGroupSpec: rayv1.HeadGroupSpec{
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{Name: "ray-head"},
```
Comment (fscnick) [inline ray-operator/controllers/ray/utils/validation_test.go:791]: Add non-empty `RayClusterSpec` on existed tests to indicate that it should not be empty. However, there is no code to enforce non-empty `RayClusterSpec` inside `ValidateRayServiceSpec()`. It has marked required in the CRD yaml.

**Comment by kevin85421** [comment]
cc @rueian

**Comment by rueian** [comment]
LGTM.

**Review by rueian — APPROVED** [review]

File: ray-operator/apis/ray/v1alpha1/rayservice_types.go [hunk 2]
```diff
@@ -54,7 +54,7 @@ type RayServiceSpec struct {
 	// Important: Run "make" to regenerate code after modifying this file
 	// Defines the applications and deployments to deploy, should be a YAML multi-line scalar string.
 	ServeConfigV2  string         `json:"serveConfigV2,omitempty"`
-	RayClusterSpec RayClusterSpec `json:"rayClusterConfig,omitempty"`
+	RayClusterSpec RayClusterSpec `json:"rayClusterConfig"`
```
Comment (MortalHappiness) [inline ray-operator/apis/ray/v1alpha1/rayservice_types.go:57]: Don't change v1alpha1 because it is no longer supported.

**Review by MortalHappiness — CHANGES_REQUESTED** [review]

**Review by MortalHappiness — APPROVED** [review]

**Review by kevin85421 — APPROVED** [review]
